rust_src/src/fns.rs

```diff
@@ -598,4 +598,53 @@ pub fn copy_sequence(mut arg: LispObject) -> LispObject {
     }
 }
 
+/// Check that ARRAY can have a valid subarray [FROM..TO),
+/// given that its size is SIZE.
+/// If FROM is nil, use 0; if TO is nil, use SIZE.
+/// Count negative values backwards from the end.
+/// Set *IFROM and *ITO to the two indexes used.
+pub fn validate_subarray_rust(
+    array: LispObject,
+    from: Option<EmacsInt>,
+    to: Option<EmacsInt>,
+    size: isize,
+) -> (EmacsInt, EmacsInt) {
+    // change from, to and size to EmacsInt
+    let int_size = size as EmacsInt;
+    let mut int_from = from.unwrap_or(0);
+    let mut int_to = to.unwrap_or(int_size);
+
+    // make negative numbers reverse array
```

Comment:
This comment is not clear to me.

```
// Negative indexes count from the end of the array.
// Adding size to them should turn them into
// equivalent positive indexes.
```

Reply:
That is better, thank you. I just pushed the changes.